test/performance/odp_l2fwd.c

```diff
@@ -721,6 +937,53 @@ static int run_worker_direct_mode(void *arg)
 	return 0;
 }
 
+static int set_pktin_vector_params(odp_pktin_queue_param_t *pktin_param, odp_pool_t vec_pool,
+				   odp_pktio_capability_t pktio_capa)
+{
+	odp_pool_info_t vec_pool_info;
+	uint64_t vec_tmo_ns;
+	uint32_t vec_size;
+
+	if (odp_pool_info(vec_pool, &vec_pool_info)) {
+		ODPH_ERR("Error: failed to get vector pool info\n");
+		return -1;
+	}
+
+	pktin_param->vector.enable = true;
+	pktin_param->vector.pool = vec_pool;
+
+	vec_size = vec_pool_info.params.vector.max_size;
+	if (vec_size > pktio_capa.vector.max_size ||
+	    vec_size < pktio_capa.vector.min_size) {
+		ODPH_ERR("Error: Invalid vector size %u for pool, valid range [%u, %u]\n",
```

Comment:
The pool vector size is not invalid per se. E.g. vector pool may be able to allocate longer vectors than packet input. I guess the code should fail here if the packet input cannot fulfill `-x, --vec_size`.

Reply:
Will add similar check here.
`        if (gbl_args->appl.vec_size == 0)
                vec_size = DEFAULT_VEC_SIZE;
        else
                vec_size = gbl_args->appl.vec_size;
`